### PR TITLE
fix(Qwen3VLCausalLMOutputWithPast): missing `hidden_states` and `atte…

### LINKED ISSUE #42594: Qwen3-VL hidden_states are not the same used to compute logits
### System Info

- `transformers` version: 4.57.1
- Platform: Linux-5.15.0-161-generic-x86_64-with-glibc2.35
- Python version: 3.10.19
- Huggingface_hub version: 0.36.0
- Safetensors version: 0.6.2
- Accelerate version: 1.12.0
- Accelerate config:    not found
- DeepSpeed version: not installed
- PyTorch version (accelerator?): 2.9.1+cu128 (NA)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using distributed or parallel set-up in script?: No

### Who can help?

_No response_

### Information

- [ ] The official example scripts
- [x] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [x] My own task or dataset (give details below)

### Reproduction

Run the following self-contained script (no access-key required):

```Python

MODEL_NAME = "Qwen/Qwen3-VL-2B-Instruct"

# Load model + processor
processor = AutoProcessor.from_pretrained(MODEL_NAME, trust_remote_code=True, cache_dir=os.environ["TRANSFORMERS_CACHE"])
model = AutoModelForImageTextToText.from_pretrained(
    MODEL_NAME,
    dtype=torch.float16,
    trust_remote_code=True,
)
model.eval()
processor = AutoProcessor.from_pretrained(MODEL_NAME)

# Create input tensors
dummy_prompt = "This is a sample prompt."
messages = [{"role": "user", "content": [{"type": "text", "text": dummy_prompt,}]}]
inputs = processor.apply_chat_template(
    messages,
    tokenize=True,
    add_generation_prompt=True,
    return_dict=True,
    return_tensors="pt"
)
inputs = inputs.to(model.device)

# Inference
outputs = model(**inputs, return_dict=True, output_hidden_states=True)
logits = model.lm_head(outputs.hidden_states[-1])
logits2 = outputs.logits

# Check if outputs are equal
assert torch.equal(logits, logits2)
```

### Expected behavior

Hello. 

I've noticed that the implementation of the [`forward`](https://github.com/huggingface/transformers/blob/v5.0.0rc0/src/transformers/models/qwen3_vl/modeling_qwen3_vl.py#L1325) method of the `Qwen3VLForConditionalGeneration` module do not explicitely outputs the `hidden_states` tuple while the [`forward`](https://github.com/huggingface/transformers/blob/v5.0.0rc0/src/transformers/models/qwen2_vl/modeling_qwen2_vl.py#L1294) method of the `Qwen2VLForConditionalGeneration` module does.

This is causing a difference between the expected logits output by the model. Specifically, the above code produce an assertion error, but when I change the `MODEL_NAME` variable to `Qwen/Qwen2-VL-2B-Instruct` the error goes away.

I believe that this can be solved by explicitly adding the `outputs.hidden_states` tuple to the [`Qwen3VLCausalLMOutputWithPast`](https://github.com/huggingface/transformers/blob/v5.0.0rc0/src/transformers/models/qwen3_vl/modeling_qwen3_vl.py#L1413) instance, exactly as it is done for the [`Qwen2VLCausalLMOutputWithPast`](https://github.com/huggingface/transformers/blob/v5.0.0rc0/src/transformers/models/qwen2_vl/modeling_qwen2_vl.py#L1399) instance. Is there a reason why they are implemented differently?

### PR DESCRIPTION
**Pull request by casinca**
# What does this PR do?

Fixes https://github.com/huggingface/transformers/issues/42594

As it was discussed in the issue, I've added the missing `hidden_states`  kwarg but also `attentions` to keep consistency with the other Qwen2 VL and 2.5 VL signatures and replaced the decorator based on what @zucchini-nlp said.

```diff
        return Qwen3VLModelOutputWithPast(
            last_hidden_state=outputs.last_hidden_state,
            past_key_values=outputs.past_key_values,
+            hidden_states=outputs.hidden_states,
+            attentions=outputs.attentions,
            rope_deltas=self.rope_deltas,
        )
```

&nbsp;

If the PR is good, I will also update the signature for `modular_qwen3_vl_moe.py`? it seems to be missing there too.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

I'm tagging @Rocketknight1 and @zucchini-nlp  since you were discussing on the issue.

### CODE REVIEW

File: src/transformers/models/qwen3_vl/modular_qwen3_vl.py [hunk 1]
```diff
@@ -1068,6 +1068,8 @@ def forward(
         return Qwen3VLModelOutputWithPast(
             last_hidden_state=outputs.last_hidden_state,
             past_key_values=outputs.past_key_values,
+            hidden_states=outputs.hidden_states,
+            attentions=outputs.attentions,
             rope_deltas=self.rope_deltas,
```
Comment (zucchini-nlp) [inline src/transformers/models/qwen3_vl/modular_qwen3_vl.py]: this is not needed, on the base model the decorator works fine

File: src/transformers/models/qwen3_vl/modular_qwen3_vl.py [hunk 2]
```diff
@@ -1080,7 +1082,7 @@ class Qwen3VLForConditionalGeneration(Qwen2_5_VLForConditionalGeneration):
     config: Qwen3VLConfig
     _checkpoint_conversion_mapping = {}
 
-    @check_model_inputs
+    @can_return_tuple
```
Comment (zucchini-nlp) [inline src/transformers/models/qwen3_vl/modular_qwen3_vl.py:1083]: nice!

**Review by zucchini-nlp — COMMENTED** [review]
Thanks, can you revert the changes on base model and then run `make fix-copies` ?

**Comment by casinca** [comment]
I've reverted for the base model and re-generated the modeling files. Concerning my initial question since `modeling_qwen3_vl_moe.py` got updated for the decorator swap, should I also add the kwargs in https://github.com/huggingface/transformers/blob/9b74e4c43402049608fc3e74c9b675d9b37ef0f7/src/transformers/models/qwen3_vl_moe/modular_qwen3_vl_moe.py#L494-L500

`make fix-copies` again and it's good?

**Comment by zucchini-nlp** [comment]
@casinca yes, you would need to update the VL-MoE model in the same way. I see a few tests failing with the MoE model rn

**Comment by casinca** [comment]
> @casinca yes, you would need to update the VL-MoE model in the same way. I see a few tests failing with the MoE model rn

I made the changes for `modular_qwen3_vl_moe.py`, all tests passed for both VL archs now

**Review by zucchini-nlp — APPROVED** [review]
Thanks, lgtm
